features/guidance/anticipate-lanes.feature

```diff
@@ -257,7 +257,116 @@ Feature: Turn Lane Guidance
+           Given the node map
+               |   |   |   | f | g |   |
+               |   |   |   |   |   |   |
+               | a | b | c | d |   | e |
+               |   |   |   |   |   |   |
+               |   |   |   | h | i |   |
+
+           And the ways
+               | nodes | turn:lanes:forward                     | name |
+               | ab    |                                        | main |
+               | bc    | left\|through\|through\|through\|right | main |
+               | cd    | left\|through\|right                   | main |
+               | de    |                                        | main |
+               | cf    |                                        | off  |
+               | ch    |                                        | off  |
+               | dg    |                                        | off  |
+               | di    |                                        | off  |
+
+          When I route I should get
+               | waypoints | route               | turns                                             | lanes                                                                                                     |
+               | a,e       | main,main,main,main | depart,use lane straight,use lane straight,arrive | ,left:false straight:false straight:true straight:false right:false,left:false straight:true right:false, |
+
+       @anticipate
+       Scenario: Anticipate Lanes for through and collapse multiple use lanes
+           Given the node map
+               |   |   | e | f | g |
+               |   |   |   |   |   |
+               | a | b | c | d |   |
+               |   |   |   |   |   |
+               |   |   | h | i | j |
+
+           And the ways
+               | nodes | turn:lanes:forward                     | name |
+               | ab    | left\|through\|through\|right          | main |
+               | bc    | left\|through\|through\|right          | main |
+               | cd    | left\|through\|through\|through\|right | main |
+               | be    |                                        | off  |
+               | bh    |                                        | off  |
+               | cf    |                                        | off  |
+               | ci    |                                        | off  |
+               | dg    |                                        | off  |
+               | dj    |                                        | off  |
+
+          When I route I should get
+               | waypoints | route          | turns                           | lanes                                                                |
+               | a,c       | main,main,main | depart,use lane straight,arrive | ,left:false straight:true straight:true right:false,                 |
+               | a,d       | main,main,main | depart,use lane straight,arrive | ,left:false straight:true straight:true right:false,                 |
```

Comment:
Yes, please have a look [here](https://github.com/Project-OSRM/osrm-backend/blob/56dad511b4148a53ff7d0be077ef715c8145105a/src/engine/guidance/lane_processing.cpp#L166-L173), the only place where anticipation now potentially collapses steps. For this I had to adapt the [collapse function](https://github.com/Project-OSRM/osrm-backend/blob/56dad511b4148a53ff7d0be077ef715c8145105a/src/engine/guidance/post_processing.cpp#L507-L525) taking lanes into account, so we only collapse if the lane data is the same.